packages/compiler/server/tmlanguage.ts

```diff
@@ -226,6 +226,17 @@ const tupleExpression: BeginEndRule = {
   patterns: [expression],
 };
 
+const arrayExpression: MatchRule = {
+  key: "array-expression",
+  scope: meta,
+  match: `(${identifier})(\\[)(\\])`,
+  captures: {
+    "1": expression,
+    "2": { scope: "punctuation.squarebracket.open.cadl" },
+    "3": { scope: "punctuation.squarebracket.close.cadl" },
+  },
+};
+
```

Comment:
We also need to update spec.emu.html for the documentation of the language grammar.